internal/server/middlewares/auth/basic.go

```diff
@@ -0,0 +1,36 @@
+package auth
+
+import (
+	"crypto/sha256"
+	"crypto/subtle"
+	"net/http"
+)
+
+type basicAuthMethod struct {
+	authDigest [32]byte
+}
+
+func newBasicAuthMethod(username, password string) *basicAuthMethod {
+	return &basicAuthMethod{
+		authDigest: sha256.Sum256([]byte(username + password)),
+	}
+}
+
+// equal returns true if another auth checker is equal.
+// This is used to deduplicate checkers for a particular route.
+func (a *basicAuthMethod) equal(other authorizationChecker) bool {
+	otherBasicMethod, ok := other.(*basicAuthMethod)
+	if !ok {
+		return false
+	}
+	return a.authDigest == otherBasicMethod.authDigest
+}
+
+func (a *basicAuthMethod) isAuthorized(_ http.ResponseWriter, request *http.Request) bool {
+	username, password, ok := request.BasicAuth()
+	if !ok {
+		return false
+	}
+	requestAuthDigest := sha256.Sum256([]byte(username + password))
+	return subtle.ConstantTimeCompare(a.authDigest[:], requestAuthDigest[:]) == 1
```

Comment:
Why do we use `subtle.ConstantTimeCompare`? Couldn't we just use

```suggestion
	return bytes.Equal(a.authDigest[:], requestAuthDigest[:])
```

Reply:
We can, constant time compare just protects against timing based attacks.
ie Brute forcing can be made easier with the timing information if we use simple equal.
See https://ropesec.com/articles/timing-attacks/

Thanks for the changes. This is my first time using golang.😂

Reply:
Oh awesome, thanks for the teaching moment! 💯

Reply:
Actually with a hash digest, I doubt it matters much. It does matter a hell lot for #2437 so that's great. keeping it just in case in this PR.

Reply:
Oh yeah that's true.